sources/zarr/large_image_source_zarr/__init__.py

```diff
@@ -684,8 +765,14 @@ def write(
             else:
                 raise TileSourceError('Output path exists (%s).' % str(path))
 
-        # TODO: compute half, quarter, etc. resolutions
-        self._validateZarr()
+        if resample is None:
+            resample = (
+                ResampleMethod.PIL_LANCZOS
+                if lossy and self.dtype == np.uint8
+                else ResampleMethod.NP_NEAREST
+            )
+
+        self._generateDownsampledLevels(resample)
```

Comment:
We probably want to do this after crop.

We also don't need to do it if we are saving via the large_image_converter, since it does its own downsampling.

Reply:
In order to generate downsampled levels after cropping, we will need to change the way we do cropping.

Currently, cropped data is kept in a separate zarr store which we create at the time of `write`. If we generate downsampled levels after this crop, we would need to propagate the cropped store to the `generateDownsampledLevels` and `addTile` methods as the target for changes. This will also change how we access `self.getMetadata()` for frames, `self.levels`, `self.sizeX`, `self.sizeY`, etc. in these methods. Since this approach would complicate other methods, I figured it would be better to change how we crop. 

Instead of managing a temporary zarr store directly, we can use a new instance of the `ZarrFileTileSource`. We can copy tiles from the original to the new sink in a similar fashion to `generateDownsampledLevels`, with a `tileIterator`. We can use the `region` arg to perform the crop. Then we can call `generateDownsampledLevels` on this instance if the output type doesn't call for the large_image_converter.

I made these changes and added a new test for this behavior in [4ef0c58](https://github.com/girder/large_image/pull/1490/commits/4ef0c584f47119f2e7c4f350255a8361430b82a0).